### PR TITLE
[WIP] Gateway API: Implement Session Persistence in BackendLBPolicy

### PR DESCRIPTION
**Pull request by gcs278**
**Please provide a description of this PR:**

Implement experimental BackendLBPolicy CRD which was introduced with Session Persistence configuration as a part of https://github.com/istio/istio/pull/53755.

WIP: Currently "works" with status and some rejection logic, but still need to clean up code and work through invalid configurations.

### CODE REVIEW

File: pilot/pkg/config/kube/gateway/conversion.go [hunk 1]
```diff
@@ -384,6 +386,155 @@ func convertGRPCRoute(r k8s.GRPCRouteRule, ctx configContext,
...
+
+func buildCookieBasedConsistentHash(backendLBPolicy *k8salpha.BackendLBPolicySpec) (*istio.LoadBalancerSettings_ConsistentHash, error) {
+	cookieName := "http-cookie"
+	if backendLBPolicy.SessionPersistence.SessionName != nil {
+		cookieName = *backendLBPolicy.SessionPersistence.SessionName
+	}
+
+	cookieTTL, err := determineCookieTTL(backendLBPolicy)
+
+	return &istio.LoadBalancerSettings_ConsistentHash{
```
Comment (howardjohn) [inline pilot/pkg/config/kube/gateway/conversion.go:486]: @costinm I thought this was supposed to translate to `envoy.http.stateful_session.cookie` -- is that right or have I not kept up with the iterations on the GEP

File: pilot/pkg/config/kube/gateway/conversion.go [hunk 2]
```diff
@@ -384,6 +386,155 @@ func convertGRPCRoute(r k8s.GRPCRouteRule, ctx configContext,
...
+			return rs
+		})
+	}
+
+	var err error
+	var drs []config.Config
+	var ancestorRefs []routeParentReference
+	for _, ref := range backendLBPolicy.TargetRefs {
+		dr := &istio.DestinationRule{
+			Host: fmt.Sprintf("%s.%s.svc.%s", ref.Name, obj.Namespace, r.Domain),
```
Comment (howardjohn) [inline pilot/pkg/config/kube/gateway/conversion.go:421]: I think this is incorrectly assuming the targetRef MUST be a Service. Or is that verified in some other way?

**Review by howardjohn — COMMENTED** [review]
Skimmed through it so far since its still in WIP, but overall looks really solid

File: pilot/pkg/config/kube/gateway/conversion.go (hunk 2 above)
Comment (gcs278) [reply pilot/pkg/config/kube/gateway/conversion.go:441]: Yep you are right. Added some logic to only accept Service targets.

**Pull request closed without merging**